rust/main/chains/hyperlane-sealevel/src/provider.rs

```diff
@@ -76,44 +150,38 @@ impl HyperlaneProvider for SealevelProvider {
     /// for all chains, not only Ethereum-like chains.
     async fn get_txn_by_hash(&self, hash: &H512) -> ChainResult<TxnInfo> {
         let signature = Signature::new(hash.as_bytes());
-        let transaction = self.rpc_client.get_transaction(&signature).await?;
 
-        let ui_transaction = match transaction.transaction.transaction {
+        let txn_confirmed = self.rpc_client.get_transaction(&signature).await?;
+        let txn_with_meta = &txn_confirmed.transaction;
+
+        let txn = match &txn_with_meta.transaction {
             EncodedTransaction::Json(t) => t,
             t => Err(Into::<ChainCommunicationError>::into(
-                HyperlaneSealevelError::UnsupportedTransactionEncoding(t),
+                HyperlaneSealevelError::UnsupportedTransactionEncoding(t.clone()),
             ))?,
         };
 
-        let received_signature = ui_transaction
-            .signatures
-            .first()
-            .ok_or(HyperlaneSealevelError::UnsignedTransaction(*hash))?;
-        let received_hash = decode_h512(received_signature)?;
-
-        if &received_hash != hash {
-            Err(Into::<ChainCommunicationError>::into(
-                HyperlaneSealevelError::IncorrectTransaction(
-                    Box::new(*hash),
-                    Box::new(received_hash),
-                ),
-            ))?;
-        }
+        Self::validate_transaction(hash, txn)?;
+        let sender = Self::sender(hash, txn)?;
+        let meta = Self::meta(txn_with_meta)?;
+        let gas_used = Self::gas(meta)?;
+        let fee = self.fee(meta)?;
+        let gas_price = Some(fee / gas_used);
```

Comment:
should we use `checked_div`?

Reply:
I added a check which reports a warning as it is done for Cosmos. Otherwise, gas price will be zero which is the edge case we accepted.